src/pages/Address/index.tsx

```diff
@@ -19,39 +25,68 @@ export const Address = () => {
   const { address } = useParams<{ address: string }>()
   const { t } = useTranslation()
   const { currentPage, pageSize } = usePaginationParamsInListPage()
+  const { tx_status: txStatus } = useSearchParams('tx_status')
 
   // REFACTOR: avoid using useSortParam
   const { sortBy, orderBy, sort } = useSortParam<'time'>(s => s === 'time')
 
+  const isPendingTxListActive = txStatus === 'pending'
+
   const addressInfoQuery = useQuery(['address_info', address], () => explorerService.api.fetchAddressInfo(address))
 
-  const addressTransactionsQuery = useQuery(
-    ['address_transactions', address, currentPage, pageSize, sort],
-    async () => {
-      try {
-        const { data: transactions, total } = await explorerService.api.fetchTransactionsByAddress(
-          address,
-          currentPage,
-          pageSize,
-          sort,
-        )
+  const listQueryKey = [
+    isPendingTxListActive ? 'address_pending_transactions' : 'address_transactions',
+    address,
+    currentPage,
+    pageSize,
+    sort,
+  ]
+  const listQueryIns = isPendingTxListActive
+    ? explorerService.api.fetchPendingTransactionsByAddress
+    : explorerService.api.fetchTransactionsByAddress
+
+  const addressTransactionsQuery = useQuery(listQueryKey, async () => {
+    try {
+      const { data: transactions, total } = await listQueryIns(address, currentPage, pageSize, sort)
+      return {
+        transactions,
+        total,
+      }
+    } catch (err) {
+      const isEmptyAddress = isAxiosError(err) && err.response?.status === 404
+      if (isEmptyAddress) {
         return {
-          transactions,
-          total,
+          transactions: [],
+          total: 0,
         }
+      }
+      throw err
+    }
+  })
+
+  const pendingTransactionCountQuery = useQuery(
+    ['address_pending_transactions', address],
+    async () => {
+      try {
+        const { total } = await explorerService.api.fetchPendingTransactionsByAddress(address, 1, 10)
```

Comment:
It seems that the backend is not handling the passed `pageSize` parameter correctly. Even when passing `10`, it still returns `page_size: 20`.

Reply:
cc @rabbitz

Reply:
https://github.com/nervosnetwork/ckb-explorer/blob/9c3faafe4250bbdb9119794d4ac4b9a012da3f8b/app/lib/fast_jsonapi/pagination_meta_generator.rb#L81

If the query result is empty, page_size returns DEFAULT_PER_PAGE = 20